webgme.js

```diff
@@ -80,6 +80,28 @@ function addToRequireJsPaths(gmeConfig) {
     addFromBasePath(gmeConfig.plugin.basePaths, 'plugin');
     addFromBasePath(gmeConfig.addOn.basePaths, 'addon');
     addFromBasePath(gmeConfig.visualization.layout.basePaths, 'layout');
+
+    // Panels are structured a little differently and may not have
+    // PANEL_NAME/PANEL_NAME.js like plugins, addons or layouts
+    var paths = gmeConfig.visualization.panelPaths,
+        panelPaths = {},
+        itemPath,
+        items;
+
+    for (var i = paths.length; i--;) {

```

Comment:
Could you refactor this into a function similar to addFromBasePath?
A minor style thing is to put the variable declarations on the top of the scope.